keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/nuts/NUTS.java

```diff
@@ -85,47 +85,45 @@ public static NUTS withDefaultConfig(KeanuRandom random) {
     private boolean saveStatistics = false;
 
     /**
-     * Sample from the posterior of a Bayesian Network using the No-U-Turn-Sampling algorithm
+     * Sample from the posterior of a probabilistic model using the No-U-Turn-Sampling algorithm
      *
-     * @param bayesNet           the bayesian network to sample from
-     * @param sampleFromVertices the vertices inside the bayesNet to sample from
+     * @param model           the probabilistic model to sample from
+     * @param variablesToSampleFrom the variables inside the probabilistic model to sample from
      * @return Samples taken with NUTS
      */
     @Override
-    public NetworkSamples getPosteriorSamples(final BayesianNetwork bayesNet,
-                                              final List<? extends Vertex> sampleFromVertices,
+    public NetworkSamples getPosteriorSamples(final ProbabilisticModel model,
+                                              final List<? extends Variable> variablesToSampleFrom,
                                               final int sampleCount) {
-        return generatePosteriorSamples(bayesNet, sampleFromVertices)
+        return generatePosteriorSamples((ProbabilisticModelWithGradient) model, variablesToSampleFrom)
```

Comment:
This cast is icky... one way to avoid it would be to parametrise `PosteriorSamplingAlgorithm` on `<? extends ProbabilisticModel>` but perhaps that's over-engineering the problem.